triggers/daml/Daml/Trigger/LowLevel.daml

```diff
@@ -244,3 +255,111 @@ data Commands = Commands
   { commandId : CommandId
   , commands : [Command]
   }
+
+-- Same as in Script; TODO factor
+data Free f a
+  = Pure a
+  | Free (f (Free f a))
+
+instance Functor f => Functor (Free f) where
+  fmap f (Pure a) = Pure (f a)
+  fmap f (Free x) = Free (fmap f <$> x)
+
+instance Functor f => Applicative (Free f) where
+  pure = Pure
+  Pure f <*> Pure a = Pure (f a)
+  Pure f <*> Free x = Free (fmap f <$> x)
+  Free x <*> my     = Free ((<*> my) <$> x)
+
+instance Functor f => Action (Free f) where
+  Pure a >>= f = f a
+  Free x >>= f = Free ((>>= f) <$> x)
+
+liftFC : Functor f => f a -> Free f a
+liftFC fa = Free (Pure <$> fa)
+
+lowerFC : Action f => Free f a -> f a
```

Comment:
Seems to be unused.

Reply:
3669c03e40041958b76d57c96ef794edcdcb4cc8